libs/common/src/helpers/change-stream/validate-audit-details.test.ts

```diff
@@ -1,49 +1,149 @@
 import { ObjectId } from 'mongodb';
-import { validateAuditDetails } from './validate-audit-details';
+import { validateAuditDetails, validateAuditDetailsAndUserType } from './validate-audit-details';
 
 describe('validateAuditDetails', () => {
-  it('throws if the input is not an object', () => {
-    expect(() => validateAuditDetails('test')).toThrow('Missing property `userType`');
+  withValidateAuditDetailsTests(validateAuditDetails);
+
+  describe.each(['tfm', 'portal'])('When userType is %s', (userType) => {
+    it('does not throw if id is a valid string', () => {
```

Comment:
Should we be explicit that it's a string representing an ObjectId rather than just valid string?

Reply:
and should we add just an it.each() of a few invalid options of invalid string/an object/null etc. throws an error?